### PR TITLE
feat(apps/legacy-api): listyieldfarming

### PR DESCRIPTION
**Pull request by eli-lim**
What this PR does / why we need it:
Implementation of legacy-api to remap the legacy API (https://api.defichain.com/v1/listyieldfarming).

Which issue(s) does this PR fixes?:
Fixes part of https://github.com/DeFiCh/jellyfish/issues/1024

### CODE REVIEW

**Comment by fuxingloh** [comment]
Conflicts

**Review by joeldavidw — DISMISSED** [review]

File: apps/legacy-api/__tests__/controllers/PoolPairController.test.ts [hunk 1]
```diff
@@ -114,3 +114,56 @@ it('/v1/listswaps', async () => {
...
+    provider_logo: 'https://defichain.com/downloads/symbol-defi-blockchain.svg',
+    provider_URL: 'https://defichain.com',
+    links: [
+      { title: 'Twitter', link: 'https://twitter.com/defichain' },
+      { title: 'YouTube', link: 'https://www.youtube.com/DeFiChain' },
+      { title: 'Reddit', link: 'https://reddit.com/r/defiblockchain' },
+      { title: 'Telegram', link: 'https://t.me/defiblockchain' },
+      { title: 'LinkedIn', link: 'https://www.linkedin.com/company/defichain' },
+      { title: 'Facebook', link: 'https://www.facebook.com/defichain.foundation' },
+      { title: 'GitHub', link: 'https://github.com/DeFiCh/ain' }
```
Comment (fuxingloh) [inline apps/legacy-api/__tests__/controllers/PoolPairController.test.ts:166]: @joeldavidw are these correct and same as DeFi Scan?

Comment (joeldavidw) [reply apps/legacy-api/__tests__/controllers/PoolPairController.test.ts:166]:
```suggestion
      { title: 'Twitter', link: 'https://twitter.com/defichain' },
      { title: 'YouTube', link: 'https://www.youtube.com/DeFiChain' },
      { title: 'Reddit', link: 'https://reddit.com/r/defiblockchain' },
      { title: 'Telegram', link: 'https://t.me/defiblockchain' },
      { title: 'LinkedIn', link: 'https://www.linkedin.com/company/defichain' },
      { title: 'Facebook', link: 'https://www.facebook.com/defichain.official' },
      { title: 'GitHub', link: 'https://github.com/DeFiCh' }
```

Comment (eli-lim) [reply apps/legacy-api/__tests__/controllers/PoolPairController.test.ts]: updated facebook and github, added discord

File: apps/legacy-api/src/controllers/PoolPairController.ts [hunk 2]
```diff
@@ -179,3 +210,40 @@ interface LegacySwapData {
   quote_volume: number
   isFrozen: 0 | 1
 }
+
+function mapPoolPairsToLegacyYieldFarmingPool (poolPair: PoolPairData): LegacyListYieldFarmingPool {
+  return {
+    // Identity
+    pair: poolPair.symbol,
+    name: poolPair.name,
+    pairLink: 'https://defiscan.live/tokens/' + poolPair.id,
```
Comment (fuxingloh) [inline apps/legacy-api/src/controllers/PoolPairController.ts:219]: Need to change this in the future when dex individual page is available.

File: apps/legacy-api/src/controllers/PoolPairController.ts [hunk 3]
```diff
@@ -179,3 +210,40 @@ interface LegacySwapData {
...
+    logo: 'https://defichain.com/downloads/symbol-defi-blockchain.svg',
```
Comment (fuxingloh) [inline apps/legacy-api/src/controllers/PoolPairController.ts:220]: Make sure this doesn't get replaced too.

**Review by joeldavidw — APPROVED** [review]

**Review by fuxingloh — APPROVED** [review]

**Pull request merged**